### PR TITLE
Reloop TerminalMix 2/4 :: mapping update

### PR DESCRIPTION
**Pull request by ronso0**
- [x] 2.1 looping UI
- [x] 2.1 FX UI with components.js
- [x] fix Library navigation
- [x] add tags to `description`so mapping can be properly sorted in GUI

Reminder for after merge:
- fix wheelTurn() to continue backspin after releasing wheel
- add mapping options for Shift + wheel turn
- add mapping options for Hotcue/Sampler button grid
- map GUI options to unused buttons (Crates, View, ... > show mixer, show waveforms etc.)
- long-press Sync to enable Master Sync

### CODE REVIEW

**Comment by ronso0** [comment]
@Be-ing How do I use a rotary (stepped, value is 63/65) encoder to control Dry/Wet of the effect units?
I'm this question came up some time ago, but I can't find it..

**Comment by Be-ing** [comment]
Implement a custom `input` function for the `dryWetKnob` Component: https://mixxx.org/wiki/doku.php/components_js#encoder

**Comment by ronso0** [comment]
Okay, in the wiki, of course. I thought there's shortcut, but that function's already pretty compact.

**Comment by Be-ing** [comment]
Yes, the 0.05 in the example on the wiki is just an example. I added a comment to the example on the wiki to explain that should be adjusted for each controller.

**Comment by Be-ing** [comment]
Did you forget to commit the JS file? I only see the XML file in this PR.

**Comment by ronso0** [comment]
I didn't commit the js file yet. I finished the looping/jump and the FX mapping and I feels good on both controllers (loop row is the same), but it was twice that I had an issue with my TerminalMix 4 and I want to make sure it's hardware related: suddenly Volume fader 2 was controlling fader 1+2 until I touched fader 1 again...

**Comment by ronso0** [comment]
In the first commit b694313 my editor changed the indendations. When reviewing after I finished this, please check the single commits for speific changes.

File: res/controllers/Reloop Terminal Mix 2-4.midi.xml [hunk 1]
```diff
@@ -1,3344 +1,3561 @@
...
-                    <Script-Binding/>
-                </options>
-            </control>
+<MixxxMIDIPreset mixxxVersion="2.1.0" schemaVersion="1">
+  <info>
+    <name>Reloop Terminal Mix 2/4 2.1</name>
+    <author>Sean M. Pappalardo (1.11), ronso0 (2.1 update)</author>
+    <description>
+A complete 4-deck preset for a single Reloop Terminal Mix 2 or 4.
+Uses scripting. Open script file and adapt variables to your needs:
```
Comment (Be-ing) [inline res/controllers/Reloop Terminal Mix 2-4.midi.xml]: Please remove "Uses scripting". This is an implementation detail that should not be presented to users. Also, I do not see the mentioned variables in the script file. Could you remove mention of those too?

Comment (ronso0) [reply res/controllers/Reloop Terminal Mix 2-4.midi.xml]: Please be patient. as I said, this is still in progress. I'm working on it right now, and of course I'll clean up the variables as well

**Comment by ronso0** [comment]
@Pegasus-RPG @Be-ing 
I just noticed the [rate range function](https://github.com/mixxxdj/mixxx/blob/master/res/controllers/Reloop%20Terminal%20Mix%202-4.js#L130) is broken:
Starting with 8% set in Preferences, I can increase the range up to 90% but I can not decrease it.
Calling `TerminalMix.pitchRange` at 90% with the fader at the limit, the rate increases but the slider moves accordingly; so touching the physical slider again shows that the range is still 90%.
I would expect the rate range to increase/decrease again step by step when reaching either 4% or 90%.

Not sure if I can fix it for 2.1. Can anyone help me with this or should I disable it for now?

**Comment by Pegasus-RPG** [comment]
The script is supposed to toggle the range up to 100% then wrap around to 8% if toggled again. Is Mixxx internally preventing setting rateRange to 100% and instead limiting it to 90%? (If so, a quick fix would just be to set the last value in TerminalMix.pitchRanges to 0.9 instead of 1.0. But I know at least one other controller script that depends on being able to set rateRange to 100%.)

**Comment by Pegasus-RPG** [comment]
Ah yes, commit 29ef9d1a70 is the culprit. That may need a more elegant fix.

**Comment by ronso0** [comment]
I finished the essentials (2.1 looping and FX interface) and it works just fine on both the TerminalMix 2 & 4. This can be reviewed and merged.
Meanwhile I'll add tags to the xml so it can be sorted reasonably in the mapping GUI, and clean up a bit.

**Comment by ronso0** [comment]
still WIP. removing duplicates, non-existant & useless controls, adding tags

**Comment by ronso0** [comment]
Done.

Will update the wiki asap.
`[Library], ChooseItem` needs to b changed to `[Library], GoToItem` as soon as #1560 is merged.

**Comment by Be-ing** [comment]
#1560 was already merged.

**Comment by ronso0** [comment]
> #1560 was already merged.

Oh...
Changes applied

**Comment by Be-ing** [comment]
LGTM thanks :+1:

**Comment by Be-ing** [comment]
Ugh somehow this broke the controller mapping validation test:
```
ControllerEngine: Watching JS File: "E:/Jenkins/workspace/v2.1/mixxx-2.1-release/architecture/amd64/platform/windows/res/controllers/Reloop Terminal Mix 2-4.js" 
ControllerEngine: Loading "E:/Jenkins/workspace/v2.1/mixxx-2.1-release/architecture/amd64/platform/windows/res/controllers/Reloop Terminal Mix 2-4.js" 
ControlDoublePrivate::getControl returning NULL for ( "[EffectRack1_EffectUnit1]" , "show_parameters" ) 
"ControllerEngine: script tried to connect to ControlObject ([EffectRack1_EffectUnit1], show_parameters) which is non-existent, ignoring." 
ControllerEngine: "Uncaught exception at line 666 in file E:/Jenkins/workspace/v2.1/mixxx-2.1-release/architecture/amd64/platform/windows/res/controllers/midi-components-0.0.js: TypeError: Result of expression 'this.showParametersConnection' [undefined] is not an object." 
ControllerEngine shutting down... 
src\test\controller_preset_validation_test.cpp(194): error: Value of: testLoadPreset(preset)
  Actual: false
Expected: true
```

**Comment by Be-ing** [comment]
Oh I think I see what's happening... I'll take care of it.

**Comment by Be-ing** [comment]
I updated the [wiki](https://mixxx.org/wiki/doku.php/reloop_terminal_mix) regarding the effects. I'm not sure if anything else on the wiki needs updating.

**Comment by ronso0** [comment]
Updating the wiki is on my ToDo for tonight or tomorrow, don't bother with it.

**Comment by ronso0** [comment]
Done. Wiki is up to date.

**Pull request merged**